src/org/pentaho/di/ui/vfs/VfsFileChooserHelper.java

```diff
@@ -54,6 +54,7 @@ public VfsFileChooserHelper( Shell shell, VfsFileChooserDialog fileChooserDialog
     this.shell = shell;
     this.variableSpace = variableSpace;
     this.fileSystemOptions = fileSystemOptions;
+    this.schemeRestrictions = new String[1];
```

Comment:
@e-cuellar is the reason this is set to 1 because of line number 133?  If so, I'd prefer this to be String[0], and add some safety on getSchemeRestriction() to check array length (because it could be 0, as set by setSchemeRestrictions).  Looks like setSchemeRestriction needs to be a little smarter too.  I'd suggest that if the API call setSchemeRestriction that we wipe the schemeRestriction and stuff it only with the incoming schemeRestriction.